### ISSUE TITLE
[Bug]: Onpress only works if layers are nested like jsx components

### ISSUE THREAD

**Issue by BubbleTrouble14** [rnmapbox/maps#3042]
### Mapbox Implementation

Mapbox

### Mapbox Version

10.7.10

### Platform

Android

### `@rnmapbox/maps` version

10.0.13

### Standalone component to reproduce

```javascript
import React from 'react';
import {
  MapView,
  ShapeSource,
  LineLayer,
  Camera,
  VectorSource,
  FillLayer,
} from '@rnmapbox/maps';

//The part which is commeted out works, triggers the on press. The other one doesnt though.

class BugReportExample extends React.Component {
  render() {
    return (
      <MapView style={{flex: 1}}>
        <Camera centerCoordinate={[-74.00597, 40.71427]} zoomLevel={14} />
        <ShapeSource
          id="polygonSource"
          onPress={() => {
            console.log('pressed');
          }}
          shape={{
            type: 'Feature',
            properties: {},
            geometry: {
              coordinates: [
                [
                  [4.029480413140874, 49.4026975280301],
                  [4.029480413140874, 46.99045121353802],
                  [11.44274825303006, 46.99045121353802],
                  [11.44274825303006, 49.4026975280301],
                  [4.029480413140874, 49.4026975280301],
                ],
              ],
              type: 'Polygon',
            },
          }}>
          {/* <FillLayer
            id="polygonFill"
            style={{fillOpacity: 0.6, fillColor: 'blue'}}
          /> */}
        </ShapeSource>
        <FillLayer
          id="polygonFill"
          sourceLayerID="polygonSource"
          sourceID="polygonSource"
          style={{fillOpacity: 0.6, fillColor: 'blue'}}
        />
      </MapView>
    );
  }
}
```

**Comment by mfazekas** [comment]
@BubbleTrouble14 pls do not use `tileUrlTemplates={["..."]}` as I won't be able to run this component as is.

**Comment by BubbleTrouble14** [comment]
Okay, ill edit it to use a geoJson

**Comment by BubbleTrouble14** [comment]
Also I am experiencing some bug after v10.0.13 no fill layers are showing anymore. Even the code above doesn't render anything. It works on ios but not on my androids. Not on the emulator or real device. Also tried the latest rc build no luck. So currenlty sticking with v10.0.13.

**Comment by BubbleTrouble14** [comment]
RNMapboxMapsImpl = "mapbox"

**Comment by mfazekas** [comment]
@BubbleTrouble14 pls correct coordinate in your example as right now the camera is different place than the polygons

**Comment by mfazekas** [comment]
@BubbleTrouble14 I've released 10.0.15-rc.0 that should fix the layer not rendering issue on android. Pls confirm that.

For the onPress only working for nested layers that's the way it works. We should document it, but I don't see it a huge deal for apps otherwise.

https://github.com/rnmapbox/maps/blob/d7936936bab2edea6cc803092c5f96bd56806d61/ios/RCTMGL-v10/RCTMGLMapView.swift#L736-L748

https://github.com/rnmapbox/maps/blob/d7936936bab2edea6cc803092c5f96bd56806d61/ios/RCTMGL-v10/RCTMGLMapView.swift#L758

https://github.com/rnmapbox/maps/blob/d7936936bab2edea6cc803092c5f96bd56806d61/ios/RCTMGL-v10/RCTMGLSource.swift#L14-L22

**Comment by BubbleTrouble14** [comment]
@mfazekas Ah makes sense, wasn't sure if it was intentional or not. Using queryRenderedFeaturesAtPoint allows me to do what I need. Thanks
